parser/src/main/kotlin/com/linkedin/dex/parser/DecodedValue.kt

```diff
@@ -0,0 +1,58 @@
+package com.linkedin.dex.parser
+
+import com.linkedin.dex.spec.DexFile
+import com.linkedin.dex.spec.EncodedValue
+import com.linkedin.dex.spec.Leb128
+
+/**
+ * A sealed class to represent the decoded values of an EncodedValue object.
+ */
+// TODO: Add support for complex types
+sealed class DecodedValue {
+    data class DecodedByte(val value: Byte) : DecodedValue()
+    data class DecodedShort(val value: Short) : DecodedValue()
+    data class DecodedChar(val value: Char) : DecodedValue()
+    data class DecodedInt(val value: Int) : DecodedValue()
+    data class DecodedLong(val value: Long) : DecodedValue()
+    data class DecodedFloat(val value: Float) : DecodedValue()
+    data class DecodedDouble(val value: Double) : DecodedValue()
+    // Value is an index into the string table
+    data class DecodedString(val value: String) : DecodedValue()
+
+    data class DecodedType(val value: String) : DecodedValue()
+    class DecodedNull : DecodedValue()
```

Comment:
We can make these null ones `object`s so that we still get proper equals & hashcode behavior